tests/integration/targets/latest/tasks/controller_managed_device_group.yml

```diff
@@ -0,0 +1,92 @@
+---
+- debug:
+    msg: "{{ nautobot_version }}"
+
+- block:
+  - name: "CONTROLLER GROUP 1: Create Group"
+    networktocode.nautobot.controller_managed_device_group:
+      url: "{{ nautobot_url }}"
+      token: "{{ nautobot_token }}"
+      name: Test Controller Group One
+      controller: controller_one
+      weight: 1000
+      state: present
+    register: test_one
+
+  - name: "CONTROLLER 1: ASSERT - Create Group"
+    assert:
+      that:
+        - test_one['changed']
+        - test_one['controller_managed_device_group']['name'] == "Test Controller Group One"
+
+  - name: "CONTROLLER GROUP 2: Create duplicate"
+    networktocode.nautobot.controller_managed_device_group:
+      url: "{{ nautobot_url }}"
+      token: "{{ nautobot_token }}"
+      name: Test Controller Group One
+      controller: controller_one
+      weight: 1000
+      state: present
+    register: test_two
+
+  - name: "CONTROLLER 2: ASSERT - Create duplicate"
+    assert:
+      that:
+        - not test_two['changed']
+        - test_two['controller_managed_device_group']['name'] == "Test Controller Group One"
+        - test_two['msg'] == "Test Controller Group One already exists"
```

Comment:
This fails because of 
```
    "diff": {
        "after": {
            "controller": "controller_one"
        },
        "before": {
            "controller": "ebe13ee7-b713-4d1b-b839-2ae0159996da"
        }
    }
```

Did I something wrong, or why is the change now with the ID?

Reply:
If the GET call returns the ID of the controller instead of the name you probably need to add controller to the [CONVERT_TO_ID](https://github.com/nautobot/nautobot-ansible/blob/develop/plugins/module_utils/utils.py#L153) dictionary so the name gets converted for proper comparison.